### PR TITLE
[iOS/Catalyst] Fix clipping issues with some views 

### LINKED ISSUE #14357: EllipseGeometry not working correctly for Labels in IOS and MacCatalyst
### Description

When creating a EllipseGeometry element inside a Label.Clip in IOS and MacOS, it only clips the text of the Label element, not the actual element itself. However in Android, it clips the entire Label element, not just the text. 

In the example below, the Text"Welcome to .NET Multi-platform App UI" is clipped in MacOS and IOS, but the actual element itself is not clipped, as it should be

MacOS 12.5.1
Screen Shot 2023-04-02 at 12 38 56 PM

IOS SE 3rd Gen, IOS 16.2

Screen Shot 2023-04-02 at 12 54 59 PM

In the Android example below, the entire label element is clipped, not just the text

Android Pixel 5 - API 33

Screen Shot 2023-04-02 at 12 49 27 PM

### Steps to Reproduce

1. Create a File > New Maui App
2. Create a label element in MainPage.xaml
3. Create a Label.Clip and EllipseGeometry inside the Label element
4. In the EllipseGeometry, set the RadiusX to 50, RadiusY to 15, and Center to 15,15
5. Run MacCatalyst or IOS simulator

### Link to public reproduction project repository

https://github.com/pschubert777/Shapes/blob/main/Shapes/MainPage.xaml

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

### Affected platform versions

IOS 16.2/15.4, MacOS 12.5.1

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix clipping issues with some views on iOS.

Captura de pantalla 2023-04-03 a las 13 01 14

### Issues Fixed

Fixes #14357

### CODE REVIEW

File: src/Core/src/Platform/iOS/WrapperView.cs [hunk 1]
```diff
@@ -122,22 +142,34 @@ public override void SetNeedsLayout()
...
 			var nativePath = path?.AsCGPath();
+
+			mask ??= MaskLayer = new CAShapeLayer();
 			mask.Path = nativePath;
+
+			var backgroundLayer = GetBackgroundLayer();
+
+			// We wrap some controls for certain visual effects like applying background gradient etc.
+			// For this reason, we have to clip the background layer as well if it exists.
+			if (backgroundLayer is null)
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/iOS/WrapperView.cs:162]: This is the main change.

File: src/Core/src/Platform/iOS/WrapperView.cs [hunk 2]
```diff
@@ -60,8 +77,8 @@ public override void LayoutSubviews()
 			if (Subviews.Length == 0)
 				return;
 
-			if (BorderView != null)
-				BringSubviewToFront(BorderView);
+			if (_borderView != null)
```
Comment (mandel-macaque) [inline src/Core/src/Platform/iOS/WrapperView.cs:80]:
```suggestion
			if (_borderView is not null)
```

File: src/Core/src/Platform/iOS/WrapperView.cs [hunk 3]
```diff
@@ -70,11 +87,14 @@ public override void LayoutSubviews()
 			if (MaskLayer != null)
 				MaskLayer.Frame = Bounds;
 
+			if (BackgroundMaskLayer != null)
```
Comment (mandel-macaque) [inline src/Core/src/Platform/iOS/WrapperView.cs:90]:
```suggestion
			if (BackgroundMaskLayer is not null)
```

File: src/Core/src/Platform/iOS/WrapperView.cs [hunk 4]
```diff
@@ -70,11 +87,14 @@ public override void LayoutSubviews()
...
+				BackgroundMaskLayer.Frame = Bounds;
+
 			if (ShadowLayer != null)
 				ShadowLayer.Frame = Bounds;
 
-			if (BorderView != null)
-				BorderView.Frame = Bounds;
+			if (_borderView != null)
```
Comment (mandel-macaque) [inline src/Core/src/Platform/iOS/WrapperView.cs:96]:
```suggestion
			if (_borderView is not null)
```

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/Core/src/Platform/iOS/WrapperView.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Core/src/Platform/iOS/WrapperView.cs:162]: We wrap some controls for certain visual effects like applying background gradient etc.
For this reason, we have to clip the background layer as well if it exists.

**Review by mandel-macaque — APPROVED** [review]

**Review by rmarinho — APPROVED** [review]

**Pull request merged**